File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    const sum: usize = (number * (number + 1)) / 2;
    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    return (number * (number + 1) * ((number * 2) + 1)) / 6;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"
#include <stdbool.h>

unsigned int sum_of_squares(unsigned int number) {
  unsigned int count = 0;
  for (unsigned int i = 0; i <= number; i++) {
    count += i * i;
  }
  return count;
}

unsigned int square_of_sum(unsigned int number) {
  unsigned int count = 0;
  for (unsigned int i = 0; i <= number; i++) {
    count += i;
  }
  return count * count;
}

unsigned int difference_of_squares(unsigned int number) {
  return square_of_sum(number) - sum_of_squares(number);
}

// int main(int argc, char *argv[]) { return EXIT_SUCCESS; }
